Rf 0.1.0

rfelsburg opened the pull request:
Initial pull request. Still work to be done in terms of fleshing out the config file. However the basics for our needs are there, using the mechanisms we'll be using in the future.

ayohrling commented on manifests/init.pp:
```diff
@@ -430,100 +432,104 @@
-
-  if $::osfamily != 'FreeBSD' {
-    file { 'snmpd.sysconfig':
+  if $use_snmpd == 'true' {
```
why not use a boolean here and make this if $use_snmpd { }

you can use validate_bool from the puppetlabs-stdlib to test this.

ayohrling commented on manifests/init.pp:
```diff
@@ -430,100 +432,104 @@
 
-  if $::osfamily == 'RedHat' {
-    file { 'snmptrapd.sysconfig':
+  if $use_snmptrapd == 'true' {
```
See above statement about booleans and not strings for boolean values.

ayohrling commented on manifests/init.pp:
```diff
@@ -430,100 +432,104 @@
-      creates => '/etc/init.d/snmptrapd',
-      require => Package['snmpd'],
-    }
+    if $::osfamily == 'RedHat' {
```
Consider using a case statement for this; nested if's on facts get messy.

ayohrling commented on manifests/init.pp:
```diff
@@ -430,100 +432,104 @@
-    require => Package['snmpd'],
-    notify  => $snmptrapd_conf_notify,
+    if $::osfamily != 'FreeBSD' {
+      file { 'snmpd.sysconfig':
```
Since you're using the osfamily fact in the actual templatename, I'm not sure you need to nest this at all. You can obfuscate teh package name out via naming/title discrepencies and the like.

ayohrling commented on manifests/params.pp:
```diff
@@ -217,6 +206,16 @@
   }
 
+  $use_snmpd = $::snmp_use_snmpd ? {
+    undef   => 'true',
```
bools vs. strings here.

ayohrling commented on templates/snmpd.conf.erb:
```diff
@@ -80,9 +101,9 @@ sysName <%= @sysname %>
 # ------------------------------------------------------------------------------
 # Dynamically Loadable Modules
-<% @dlmod.each do |mod| -%>
+  <% @dlmod.each do |mod| -%>
```
Indent the code inside the blocks here, not the whole block itself or the whitespace may show up in teh file.

Commit:
it starts.

Commit:
rewrite of initial code

Commit:
Moved installation of package into install.pp

Commit:
Moved service changes to service.pp
VAlidate input in init.pp

Commit:
Enable snmpd, disable snmp client

Commit:
stupiud braces

Commit:
You can't set parent variables dumbass

Commit:
stupid
stupid
stupid
stupid

Commit:
linting

Commit:
it's always the little things

Commit:
not a boolean

Commit:
misspelling

Commit:
just because I'm out of ideas

Commit:
scope changes

Commit:
scoping

Commit:
module files

Commit:
adding in required fields

Commit:
snmp package name is wonky.

Commit:
scoping again?

Commit:
scoping back to normal. stupid troubleshooting

Commit:
starting the config section

added config.pp
modded init.pp to add in new vars
modded params to add in new vars
new template, naming for future proofing different distros

Commit:
type snmp vs snmpd

Commit:
Starting work with concat to manage the snmp v3 user file. Testing basic setup, will
most likely have to couple this with something else, as snmp will mod this file.

Commit:
enclosing variable in braces

Commit:
requires concat::fragment, regardless of documentation. Seems like change via puppet4

Commit:
testing whether this is the initial problem with the exec

Commit:
dont replace the entire file, just append

Commit:
mayhaps concat is not our best bet

Commit:
snmpd/snmptrapd needing to be stopping order to add a user makes things difficult.

longterm we will probably end up with a custom resource, but for now, exec is the least
complex way to start this. getting the class and vars setup first

Commit:
starting with basic functionality required by snmpd

Commit:
helps if you set the vars

Commit:
silly scope, variables are for parents

Commit:
before "fixing" anything else.

Commit:
testing base function

Commit:
/sigh

Commit:
define not class

Commit:
Initial test

Commit:
fixing values

Commit:
which oh which to escape

Commit:
maybes?

Commit:
variable name

Commit:
knew it

Commit:
hmmm

Commit:
stupid letters

Commit:
escaping is hard

Commit:
Cleaning up variables.

Commit:
testing new mechnaism to pull user directly into template, rather than tacking on

Commit:
It's not supposed to be astrin you loony linter. And robots don't say ye.

Commit:
ye

Commit:
I don't want to talk about it

Commit:
if the statement is bad

Commit:
abstract the users to their own file.

Commit:
abstraction of the users to a seperate file.

Commit:
external config directory management as well

Commit:
conditionals require a condition

Commit:
/sigh

Commit:
I still don't want to talk about it

Commit:
Clearing up linting

Commit:
To shut up lint

Commit:
shutup

Commit:
it's always the little things, like correctly coding, and not making stupid mistakes, that get you

Commit:
I hate scope, everyone knows listerine is better

Commit:
grumble grumble grumble

Commit:
hmmm

Commit:
not needed for testing now

ayohrling commented on manifests/config.pp:
```diff
@@ -0,0 +1,13 @@
+    owner   => 'root',
+    group   => 'root',
+    path    => $::snmp::snmpd_config_file,
+    content => template('snmp/RedHat/snmpd/snmpd.conf.erb'),
```
if you have this nested under RedHat say, you'd probably want to use a fact here? like template("${module_name}/${::osfamily}/snmpd/snmpd.conf.erb"),

rfelsburg replied:
I agree, I didn't use it at the moment, because there's a whole bunch more that should be added to the  config file as well for the template. This is just our basics.

ayohrling commented on manifests/init.pp:
```diff
@@ -0,0 +1,40 @@
+  $snmpd_service_name               = $::snmp::params::snmpd_service_name,
+
+  $snmpd_config_file                = $::snmp::params::snmpd_config_file,
+
```
nitpicky; but empty lines in the parameters? usually OK for comments about hte params to be there, but empty lines should probably be pulled; can be done later.

ayohrling replied:
Also the = should be 1 space after the longest parameter name; i.e they all should align at $snmpd_config_agentaddress = $::snmp....

ayohrling commented on manifests/install.pp:
```diff
@@ -0,0 +1,34 @@
+  }
+
+  # validate input for snmp
+  if ! ($::snmp::snmp_package_ensure in ['present', 'absent', 'latest']) {
```
This could theoretcally be used for specific versioning as well; so you could add a clauses here to match it to versioning regex that would meet requirements of a version of snmpd? Or just allow this to be any string value w/ validate string from the init class and remove this block entirely.

ayohrling commented on manifests/params.pp:
```diff
@@ -0,0 +1,20 @@
+# SNMP Class to set default options
+class snmp::params {
+  $snmpd_package_ensure             = 'present'
```
same nitpicky comment here w/ the spaces for the = sign alignment

ayohrling commented on manifests/params.pp:
```diff
@@ -0,0 +1,20 @@
+  $snmp_package_name                = 'net-snmp-utils'
+
+  $snmpd_service_ensure             = 'running'
+  $snmpd_service_enable             = 'true'
```
make this a boolean and see below

ayohrling commented on manifests/service.pp:
```diff
@@ -0,0 +1,31 @@
+    fail('snmpd_package_ensure parameter must be either running, or stopped')
+  }
+
+  if ! ($::snmp::snmpd_service_enable in ['true', 'false', 'manual', 'mask']) {
```
f ((! $variable in [‘bleh’]) || $variable.is_a(Boolean)) { fail() }

ayohrling commented on manifests/user.pp:
```diff
@@ -0,0 +1,87 @@
+  $auth_password  = undef,
+  $priv_enc_type  = 'AES',
+  $priv_password  = undef,
+
```
nitpicky empty line comment

ayohrling commented on manifests/user.pp:
```diff
@@ -0,0 +1,87 @@
+  exec { 'stop_snmpd':
+    path    => '/bin:/sbin:/usr/bin:/usr/sbin',
+    command => "service ${snmpd_service_name} stop; sleep 5",
+    user    => 'root',
```
use refreshonly here or this will run every puppet run

ayohrling commented on manifests/user.pp:
```diff
@@ -0,0 +1,87 @@
+    path    => '/bin:/sbin:/usr/bin:/usr/sbin',
+    command => "service ${snmpd_service_name} start",
+    user    => 'root',
+    require => Exec['stop_snmpd'],
```
use refreshonly here or this will run everytime

ayohrling reviewed (changes requested):
Some nitpicky; some legit

Commit:
Update user defined class

This change updates the user defined class to write a random string file
out based on a seed of the user's passwords. This is also pseudo-random
based on the fqdn which means it's random but it won't change between
Puppet runs. That allows us to use it deterministically to evaluate if
the user's password needs to be changed so we're not stopping/starting the
service every 30 minutes for every user.

Also fixed some random things here and there and moved the resources that
are static in the user defined type out to the config class. Ordered the config
class before the Snmp::User being evaluated as well.

Commit:
styling/logic/flow

Commit:
oid values for determining which auth priv values are being used

Commit:
variable naming?

Commit:
testing

Commit:
it really doesn't like numerical variable names not in quotes

Commit:
stupid notice

Commit:
grumble

Commit:
dynamics

Commit:
bash would've done it.

Commit:
reverting back

Commit:
bah

rfelsburg committed:
Merge branch 'user_changes' into rf_0.1.0

Commit:
commas

Commit:
Moved logic out, instead of requiring a string for the true false, just check for both.

Commit:
might as well keep the string checks in

Commit:
Oh get over it puppet-lint, true is technically a string as well.

Commit:
just has to be different didn't you puppet

Commit:
not checking a variable value, but the variable types itself

Commit:
not sure what it's complaining about, maybe not like the negation?

Commit:
hmm

Commit:
original

Commit:
no or

Commit:
missing a paren.

Commit:
shutup adam

Commit:
stringify all the things

Commit:
Adam did it.
